### PR TITLE
SDL_GetMouseState: More detail on buttons, and use cases.

### PR DESCRIPTION
**Pull request by SDLWikiBot**
This edit was made by @RPBCACUEAIIBH.

Live page is here: https://wiki.libsdl.org/SDL_GetMouseState

If this user should be blocked from further edits, an admin should go to https://wiki.libsdl.org/RPBCACUEAIIBH/block
If this user should be trusted to make direct pushes to main, without a pull request, an admin should go to https://wiki.libsdl.org/RPBCACUEAIIBH/trust

### CODE REVIEW

File: SDL_GetMouseState.mediawiki [hunk 1]
```diff
@@ -31,6 +32,8 @@ button), and <code>x</code> and <code>y</code> are set to the mouse cursor
 position relative to the focus window. You can pass NULL for either
 <code>x</code> or <code>y</code>.
 
+You may wanna use [[SDL_GetGlobalMouseState]]() for multi monitor setup, since x and Y drop to 0 when passing from 1 monitor to the next, thus you don't know which monitor the mouse is on...
```
Comment (slouken) [inline SDL_GetMouseState.mediawiki:35]: The coordinates are window relative for this function.

Comment (slouken) [reply SDL_GetMouseState.mediawiki:35]: SDL_GetGlobalMouseState() should return the extended buttons in the latest snapshot, can you verify?
https://www.libsdl.org/tmp/SDL-2.0.zip

Comment (RPBCACUEAIIBH) [reply SDL_GetMouseState.mediawiki:35]: I downloaded unpacked and did:
./configure
make
sudo make install
in the unpacked directory, and it finished without problems. Then I completely rebuilt my application and it does the same with SDL_GetGlobalMouseState as it did before. Still only recognizes 3 of 5 buttons.

Comment (slouken) [reply SDL_GetMouseState.mediawiki:35]: Is the problem maybe that the XQueryPointer call here doesn't return extended button state?
https://github.com/libsdl-org/SDL/blob/main/src/video/x11/SDL_x11mouse.c#L403

Comment (slouken) [reply SDL_GetMouseState.mediawiki:35]: Yup, that's the problem...

Okay, try this?
https://www.libsdl.org/tmp/SDL-2.0.zip (updated file)

Comment (RPBCACUEAIIBH) [reply SDL_GetMouseState.mediawiki:35]: This is the output of SDL_GetMouseState
00000:00:15.887217 [ Trace  1000 -->   Debug ]: X 364 Y 181 Buttons 1
00000:00:15.961068 [ Trace  1000 -->   Debug ]: X 364 Y 181 Buttons 0
00000:00:17.267401 [ Trace  1000 -->   Debug ]: X 364 Y 181 Buttons 2
00000:00:17.431823 [ Trace  1000 -->   Debug ]: X 364 Y 181 Buttons 0
00000:00:18.150384 [ Trace  1000 -->   Debug ]: X 364 Y 181 Buttons 4
00000:00:18.265462 [ Trace  1000 -->   Debug ]: X 364 Y 181 Buttons 0
00000:00:20.231536 [ Trace  1000 -->   Debug ]: X 364 Y 181 Buttons 8
00000:00:20.391225 [ Trace  1000 -->   Debug ]: X 364 Y 181 Buttons 0
00000:00:23.179676 [ Trace  1000 -->   Debug ]: X 364 Y 181 Buttons 16
00000:00:23.448986 [ Trace  1000 -->   Debug ]: X 364 Y 181 Buttons 0

Comment (RPBCACUEAIIBH) [reply SDL_GetMouseState.mediawiki:35]: Just to confirm, I'm on Ubuntu 20.10 (a bit outdated...) Windowing system: X11

**Comment by RPBCACUEAIIBH** [comment]
hmm... I tried mocking around with it just to make sure... testing values...
this also doesn't work...
`                    buttons |= (mask & Button1Mask) ? SDL_BUTTON_LMASK : 0;
                    buttons |= (mask & Button2Mask) ? SDL_BUTTON_MMASK : 0;
                    buttons |= (mask & Button3Mask) ? SDL_BUTTON_RMASK : 0;
                    /* Use the SDL state for the extended buttons - it's better than nothing */
//                    buttons |= (SDL_GetMouseState(NULL, NULL) & (SDL_BUTTON_X1MASK|SDL_BUTTON_X2MASK));
                    buttons = 0;
                    buttons |= 8;
                    buttons |= 16;`

it does the same as before... buttons 1-3 work, 4-5 does not... it should give me 24 regardless which button I press... so I don't know what's the problem... :S

Should I have uninstalled the old version of SDL before sudo make install?

**Comment by slouken** [comment]
> Should I have uninstalled the old version of SDL before sudo make install?

Yes, definitely.

**Comment by RPBCACUEAIIBH** [comment]
I just did, and I think I'll have to move to my other PC... build failed... multiple missing dependencies, that I can't re-install cause support ended... :S I think it's time for a fresh OS install on this one...

**Comment by RPBCACUEAIIBH** [comment]
Unfortunately I could not get my application to work on my other machine, which is odd cause It worked before... an update broke it perhaps??? Now it's the SDL2_Mixer library throwing an error: "SDL Audio initialization failed: dsp: No such audio device", and I just about had enough of troubleshooting for today...
BTW it's ubuntu 21.10 with wayland instead of X11 anyway, so not sure if it would be good for testing X11 issue anyway...

**Comment by icculus** [comment]
I'm assuming this is resolved, so I'm closing this. Let me know if we need to keep an issue open for this, though!

**Pull request closed without merging**